navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/camera/NavigationCamera.java

```diff
@@ -0,0 +1,195 @@
+package com.mapbox.services.android.navigation.ui.v5.camera;
+
+import android.content.Context;
+import android.location.Location;
+import android.support.annotation.NonNull;
+import android.util.DisplayMetrics;
+import android.util.SparseArray;
+
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.services.Constants;
+import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation;
+import com.mapbox.services.android.navigation.v5.routeprogress.ProgressChangeListener;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
+import com.mapbox.services.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.services.api.utils.turf.TurfConstants;
+import com.mapbox.services.api.utils.turf.TurfMeasurement;
+import com.mapbox.services.commons.geojson.LineString;
+import com.mapbox.services.commons.models.Position;
+
+import timber.log.Timber;
+
+import static android.content.res.Configuration.ORIENTATION_LANDSCAPE;
+import static android.content.res.Configuration.ORIENTATION_PORTRAIT;
+
+public class NavigationCamera implements ProgressChangeListener {
+
+  private static final int CAMERA_TILT = 57;
+  private static int CAMERA_ZOOM = 17;
+
+  private MapboxMap mapboxMap;
+  private MapboxNavigation navigation;
+
+  private Location location;
+  private double targetDistance;
+  private boolean trackingEnabled = true;
+
+  public NavigationCamera(@NonNull Context context, @NonNull MapboxMap mapboxMap,
+                          @NonNull MapboxNavigation navigation) {
+    this.mapboxMap = mapboxMap;
+    this.navigation = navigation;
+    initialize(context);
+  }
+
+  public void start(DirectionsRoute route) {
+    if (route != null) {
+      CameraPosition cameraPosition = buildCameraPositionFromRoute(route);
+      mapboxMap.animateCamera(CameraUpdateFactory.newCameraPosition(cameraPosition), 2000,
+        new MapboxMap.CancelableCallback() {
+          @Override
+          public void onCancel() {
+
```

Comment:
What happens when the user moves the map as the camera first starts the animation? does the progressChange still get added?